### PR TITLE
NEP-292 : article category weight added on the basis of read articles

### PR DESCRIPTION
**Pull request by ShivaAryal**
-   [x] Include Jira Story No. in Title
-   [ ] Write Unit Tests
-   [ ] Add screenshots whenever possible
-   [x] Make sure Github Action build succeed
-   [x] Address Automated Code Reviews from LGTM and Deepscan
-   [ ] Add Reviewers
-   [ ] Force Merge PRs if not approved for testing in time

### CODE REVIEW

File: src/database/resolvers.js [hunk 1]
```diff
@@ -187,19 +189,24 @@ module.exports = {
 		},
 
 		saveReadArticle: async (parent, args, {}) => {
-			const {
-				input: { nid, articleId }
+			let {
+				input: { nid, articles }
 			} = args
-
+			articles = JSON.parse(JSON.stringify(articles))
```
Comment (syuraj) [inline src/database/resolvers.js:195]: it seems unnecessary to stringify & parse again?

File: src/db-service/UserDbService.js [hunk 2]
```diff
@@ -9,6 +10,27 @@ module.exports = {
 	removeUnRegisteredUser: async (fcmToken) => {
 		const response = await User.findOneAndUpdate({fcmToken},{status: 'inactive'},{useFindAndModify: false})
 		return response
+	},
+
+	calculateUserSpecificWeight: async (nid) => {
```
Comment (syuraj) [inline src/db-service/UserDbService.js:15]: can you rename nid to more descriptive word?

File: src/db-service/UserDbService.js [hunk 3]
```diff
@@ -9,6 +10,27 @@ module.exports = {
...
+		if(nid){
```
Comment (syuraj) [inline src/db-service/UserDbService.js:16]: it is generally preferred to return early for invalid scenario, so

```
if(!nid){
return []
}

//rest of the code
const userReadArticles = 
```

**Review by syuraj — CHANGES_REQUESTED** [review]

File: src/db-service/UserDbService.js (hunk 2 above)
Comment (ShivaAryal) [reply src/db-service/UserDbService.js:15]: nid suru bata nai user schema haru ma thiyo so tehi continue gareko, sabai change garam ra?

**Pull request merged**